source/tests/system/nidaqmx_driver_api_tests.cpp

```diff
@@ -1421,31 +1569,34 @@ TEST_F(NiDAQmxDriverApiTests, SelfCal_Succeeds)
 
 TEST_F(NiDAQmxDriverApiTests, AddNetworkDeviceWithInvalidIP_ErrorRetrievingNetworkDeviceProperties)
 {
-  try {
-    auto response = AddNetworkDeviceResponse{};
-    add_network_device("0.0.0.0", response);
-    FAIL() << "We shouldn't get here.";
-  }
-  catch (const nidevice_grpc::experimental::client::grpc_driver_error& ex) {
-    expect_driver_error(ex, RETRIEVING_NETWORK_DEVICE_PROPERTIES_ERROR);
-  }
+  EXPECT_THROW({
+    try {
+      auto response = AddNetworkDeviceResponse{};
+      add_network_device("0.0.0.0", response);
+    }
+    catch (const nidevice_grpc::experimental::client::grpc_driver_error& ex) {
+      expect_driver_error(ex, RETRIEVING_NETWORK_DEVICE_PROPERTIES_ERROR);
+      throw;
+    }
+  }, nidevice_grpc::experimental::client::grpc_driver_error);
 }
 
 TEST_F(NiDAQmxDriverApiTests, ConfigureTEDSOnNonTEDSChannel_ErrorTEDSSensorNotDetected)
 {
-  try {
-    client::configure_teds(stub(), AI_CHANNEL, "");
-    FAIL() << "We shouldn't get here.";
-  }
-  catch (const nidevice_grpc::experimental::client::grpc_driver_error& ex) {
-    expect_driver_error(ex, TEDS_SENSOR_NOT_DETECTED_ERROR);
-  }
+  EXPECT_THROW({
+    try {
+      client::configure_teds(stub(), AI_CHANNEL, "");
+    }
+    catch (const nidevice_grpc::experimental::client::grpc_driver_error& ex) {
+      expect_driver_error(ex, TEDS_SENSOR_NOT_DETECTED_ERROR);
+      throw;
+    }
+  }, nidevice_grpc::experimental::client::grpc_driver_error);
 }
 
 TEST_F(NiDAQmxDriverApiTests, HardwareTimedTask_WaitForNextSampleClock_Succeeds)
 {
   create_ao_voltage_chan(0.0, 10.0);
-  create_ai_voltage_chan(0.0, 10.0);
```

Comment:
This fixes:
```
[ RUN      ] NiDAQmxDriverApiTests.HardwareTimedTask_WaitForNextSampleClock_Succeeds
unknown file: error: C++ exception with description "Task cannot contain a channel with the specified channel type, because the task already contains channels with a different channel type.
Create one task for each channel type.
Virtual Channel Name: ao0
I/O Type Required for Virtual Channel: DAQmx_Val_AO
Physical Channel Name: gRPCSystemTestDAQ/ai0
I/O Type Required for Virtual Channel: DAQmx_Val_AI

Task Name: _unnamedTask<2C>

Status Code: -200559" thrown in the test body.
[  FAILED  ] NiDAQmxDriverApiTests.HardwareTimedTask_WaitForNextSampleClock_Succeeds (17 ms)
```